types/math/dec_test.go

```diff
@@ -0,0 +1,78 @@
+package math
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDec(t *testing.T) {
```

Comment:
Yup, this looks like a good spot for some property tests :D

Even if we don't add those in this PR, perhaps it would be good to identify some properties we'd expect these types to have? They can just be in comments in some basic mathematical notation for now.

Reply:
@ruhatch  Just added some suggestions as comments in the bottom of `TestDec()`. I'm pretty new to property tests so let me know what you think?

If the decimal nature of this arithmetic means that actual equality isn't really feasible to test, then perhaps we can property test to keep error range as minimal as possible? e.g. `(x/y)*y - x) < 1e-15` or something similar?

Reply:
Yeah, some error range property would be great. I'm not sure what the standard should be, as I'm not super experienced with float stuff!